api.py

```diff
@@ -71,14 +122,126 @@ def get(self):
 		self.response.headers['Content-Type'] = 'application/json'
 		self.response.write(json.dumps(output))
 
+class RankAPI(webapp2.RequestHandler):
+	def get(self, id):
+		if not check_authentication(self):
+			return
+
+		member = Member.query(Member.id == id).get()
+		if not member:
+			# 404 if a nonexistent member is specified
+			self.error(404)
+			return
+
+		selected_semester = self.request.get('semester')
+		
+		if not selected_semester:
+			selected_semester = get_current_semester()
+		else:
+			# Avoid throwing error 500 if a bad semester string is supplied
+			try:
+				selected_semester = float(selected_semester)
+			except ValueError:
+				self.error(400)
+				return
+		#Build output dict with all rank related results
+		output = {
+		'rankName': member.get_rank_name(selected_semester),
+		'rankDisp': member.get_rank_disp(selected_semester),
+		'rankWithName': member.get_name_with_rank(selected_semester)}
```

Comment:
Since the API is to fetch the **rank**, not the member, I am inclined to call the rank fields, `name`, `abbr`, and `disp` (for consistency with `ranks.py`), and then if we want to include name-with-rank, give that a name that suggests the member name is being appended to the rank, not the other way around.

Reply:
Agree, will rename to be more consistent. Is rankName good for signifying we are appending the name?

Also I realized looking through this right now that there is no method to access `abbr` in `ranks.py`, is it worth adding in this commit?